sdk/clientcore/tools/annotation-processor/src/main/java/io/clientcore/tools/codegen/utils/PathBuilder.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.tools.codegen.utils;
+
+import io.clientcore.tools.codegen.exceptions.MissingSubstitutionException;
+import io.clientcore.tools.codegen.models.HttpRequestContext;
+import io.clientcore.tools.codegen.models.Substitution;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class PathBuilder {
+    // this class takes a 'raw host' string that contains {} delimited parameters, and needs to convert it into a
+    // Java string concatenation that can be used in the generated code. For example, the raw host string:
+    // https://{endpoint}/keys/{path1}
+    // would be converted into:
+    // "https://" + endpointParam + "/keys/" + pathValue
```

Comment:
Something that needs to be handled is where the value is null. The logic in `SwaggerMethodParser` is `null -> ""`. So, we'll probably want

```suggestion
    // https://{endpoint}/keys/{path1}
    // would be converted into:
    // "https://" + Objects.toString(endpointParam, "") + "/keys/" + Objects.toString(pathValue, "")
```

Or, we can choose to break this behavior and have `null -> "null"`